### PR TITLE
deploy image-registry on infra nodes

### PR DESCRIPTION
**Pull request by boranx**
implements: OSD-2392

### CODE REVIEW

**Comment by cblecker** [comment]
/hold
This has some issues. Discussing with @boranx on Slack

File: deploy/osd-registry/registry-infraNodes.patch.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: imageregistry.operator.openshift.io/v1
+applyMode: AlwaysApply
+kind: Config
+name: cluster
+# patch registry to deploy on infras: https://docs.openshift.com/container-platform/4.2/machine_management/creating-infrastructure-machinesets.html#infrastructure-moving-registry_creating-infrastructure-machinesets
+patch: '{"spec":{"nodeSelector":{"node-role.kubernetes.io/infra":""},"tolerations":[{"effect":"NoSchedule","key":"node-role.kubernetes.io/infra","operator":"Exists"}]}'
```
Comment (jewzaam) [inline deploy/osd-registry/registry-infraNodes.patch.yaml]:
```suggestion
patch: '{"spec":{"nodeSelector":{"node-role.kubernetes.io/infra":""},"tolerations":[{"effect":"NoSchedule","key":"node-role.kubernetes.io/infra","operator":"Exists"}]}}'
```

Missing closing bracket.

File: hack/00-osd-managed-cluster-config.selectorsyncset.yaml.tmpl [hunk 2]
```diff
@@ -3194,6 +3194,12 @@ objects:
       name: cluster
       patch: '{"spec":{"defaultRoute":true}}'
       patchType: merge
+    - apiVersion: imageregistry.operator.openshift.io/v1
+      applyMode: AlwaysApply
+      kind: Config
+      name: cluster
+      patch: '{"spec":{"nodeSelector":{"node-role.kubernetes.io/infra":""},"tolerations":[{"effect":"NoSchedule","key":"node-role.kubernetes.io/infra","operator":"Exists"}]}'
```
Comment (jewzaam) [inline hack/00-osd-managed-cluster-config.selectorsyncset.yaml.tmpl]:
```suggestion
      patch: '{"spec":{"nodeSelector":{"node-role.kubernetes.io/infra":""},"tolerations":[{"effect":"NoSchedule","key":"node-role.kubernetes.io/infra","operator":"Exists"}]}}'
```

File: deploy/osd-registry/registry-infraNodes.patch.yaml (hunk 1 above)
Comment (jewzaam) [reply deploy/osd-registry/registry-infraNodes.patch.yaml]: Note you can test the json with

`oc patch config cluster --type merge -p '<json here>'`

**Comment by boranx** [comment]
@jewzaam can you review again?

**Comment by cblecker** [comment]
[test]

**Pull request merged**